### PR TITLE
Refactor cluster controller

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Refactoring to have most of the methods in a better describing file. Fixes https://github.com/FoundationDB/fdb-kubernetes-operator/issues/547 there are only a few methods that I want to revisit once this PR is merged but this one is huge enough already to merge that stepwise.

## Type of change

*Please select one of the options below.*

- Bug fix (non-breaking change which fixes an issue) / Refactoring

# Discussion

I moved most of the methods into the `internal` package to reduce the methods we expose.

# Testing

Unit tests and locally

# Documentation

None.

# Follow-up

Once this PR is merged I take a second look to ensure we only expose what we want to expose.

### CODE REVIEW

**Comment by brownleej** [comment]
Do we want to look at the exposed methods in the other packages as part of #547?

File: controllers/add_pods.go [hunk 1]
```diff
@@ -35,7 +37,7 @@ type AddPods struct{}
 
 // Reconcile runs the reconciler's work.
 func (a AddPods) Reconcile(r *FoundationDBClusterReconciler, context ctx.Context, cluster *fdbtypes.FoundationDBCluster) *Requeue {
```
Comment (brownleej) [inline controllers/add_pods.go:39]: Should we consider moving the subreconcilers into the internal package? Having them in the public package means that we wouldn't be able to fully remove one without a major release.

Comment (brownleej) [reply controllers/add_pods.go:39]: If we moved them into the internal package we could also look into creating subpackages for different reconcilers to keep the code more organized.

File: controllers/cluster_controller.go [hunk 2]
```diff
@@ -45,24 +39,19 @@ import (
...
-
 // FoundationDBClusterReconciler reconciles a FoundationDBCluster object
 type FoundationDBClusterReconciler struct {
 	client.Client
 	Recorder            record.EventRecorder
 	Log                 logr.Logger
 	InSimulation        bool
 	PodLifecycleManager PodLifecycleManager
-	PodClientProvider   func(*fdbtypes.FoundationDBCluster, *corev1.Pod) (FdbPodClient, error)
+	PodClientProvider   func(*fdbtypes.FoundationDBCluster, *corev1.Pod) (internal.FdbPodClient, error)
```
Comment (brownleej) [inline controllers/cluster_controller.go:54]: If this field refers to an internal type, does it mean that something that imports the `controllers` package would be unable to create a `FoundationDBClusterReconciler`?

File: internal/process_config.go [hunk 3]
```diff
@@ -0,0 +1 @@
+package internal
```
Comment (brownleej) [inline internal/process_config.go:1]: What is this file for?

**Review by brownleej — CHANGES_REQUESTED** [review]

File: controllers/add_pods.go (hunk 1 above)
Comment (johscheuer) [reply controllers/add_pods.go:39]: Sounds good, I would prefer to open a follow up issue for that and move the subreconcilers in another PR. This PR is already touching too many files :) https://github.com/FoundationDB/fdb-kubernetes-operator/issues/827

File: internal/process_config.go (hunk 3 above)
Comment (johscheuer) [reply internal/process_config.go:1]: Oh, I guess I wanted to move something here and then didn't use it. I'll remove it!

**Review by johscheuer — COMMENTED** [review]
I just noted that I have to add the Apache header at some places.

File: controllers/cluster_controller.go (hunk 2 above)
Comment (johscheuer) [reply controllers/cluster_controller.go:54]: Oh right, I'll check that. Good point 👍

**Comment by johscheuer** [comment]
> Do we want to look at the exposed methods in the other packages as part of #547?

I would look at the other packages in another PR/step to have multiple refactoring PRs instead of one huge.

File: controllers/cluster_controller.go (hunk 2 above)
Comment (johscheuer) [reply controllers/cluster_controller.go:54]: I'll created a method for that, which only exposes fields that should be changed. We could also think about exposing a smaller surface before 1.0 but I would solve that in another PR.

**Review by 09harsh — APPROVED** [review]

**Pull request merged**